opentelemetry-api/src/global/metrics.rs

```diff
@@ -72,8 +75,8 @@ pub fn meter_provider() -> GlobalMeterProvider {
 /// If the name is an empty string, the provider will use a default name.
 ///
 /// This is a more convenient way of expressing `global::meter_provider().meter(name, None, None)`.
-pub fn meter(name: &'static str) -> Meter {
-    meter_provider().versioned_meter(name, None, None)
+pub fn meter(name: impl Into<Cow<'static, str>>) -> Meter {
+    meter_provider().versioned_meter(name.into(), None, None)
```

Comment:
Nit: Can we use ` meter_provider().meter(name.into())` to avoid the extra `None`s?